MeshLib/MeshQuality/MeshValidation.cpp

```diff
@@ -147,4 +150,49 @@ MeshValidation::ElementErrorCodeOutput(const std::vector<ElementErrorCode> &erro
 	return output;
 }
 
+unsigned MeshValidation::detectHoles(MeshLib::Mesh const& mesh)
+{
+	MeshLib::Mesh* boundary_mesh (MeshSurfaceExtraction::getMeshBoundary(mesh));
+	std::vector<MeshLib::Element*> const& elements (boundary_mesh->getElements());
+
+	std::vector<unsigned> sfc_idx (elements.size(), std::numeric_limits<unsigned>::max());
+	unsigned current_surface_id (0);
+	std::vector<unsigned>::const_iterator it = sfc_idx.cbegin();
+
+	while (it != sfc_idx.cend())
+	{
+		int const idx = static_cast<int>(std::distance(sfc_idx.cbegin(), it));
```

Comment:
okay, i removed the error code now.
I know there was a reason I put it there but I cannot reproduce any problems now and it *might* have been necessary at some point during implementation. anyway, it's gone now ...